lib/rack/session/cookie.rb

```diff
@@ -102,11 +134,43 @@ def encode(str); str; end
         def decode(str); str; end
       end
 
-      attr_reader :coder
+      class Marshal
+        def encode(str)
+          ::Marshal.dump(str)
+        end
+
+        def decode(str)
+          ::Marshal.load(str) if str
+        end
+      end
+
+      attr_reader :coder, :encryptors
 
       def initialize(app, options = {})
-        @secrets = options.values_at(:secret, :old_secret).compact
-        @hmac = options.fetch(:hmac, OpenSSL::Digest::SHA1)
+        # TODO handle decoding the secrets... are they base64? hex? raw binary
```

Comment:
We need to consider how we want to handle the `:secrets` option. If this option is encoded as base64 or hex, we need to decode. If these encoded values are not handled correctly, key material can be mishandled.

Reply:
I think we should document that secrets are expected to be raw.  If users want to use base64 or hex, they should decode before passing the option to the middleware.

Reply:
Sounds good, I can make changes as needed.

I think the documentation just needs to emphasis the importance of how to handle the secret key(s). Often, these values are coming from environment variable or configuration files are very likely to be encoded somehow.

My other thought on this was we can have `:secrets` handle raw keys and add `:base64_secrets` and `:hex_secrets` which imply a decoding step before passing the keys off to the Encryptor.